### PR TITLE
[build] Add an action to check API/ABI compatibility

### PR DESCRIPTION
**Pull request by yomnes0**
Add an action to check backward API/ABI compatibility

### CODE REVIEW

File: .github/workflows/abi.yml [hunk 1]
```diff
@@ -0,0 +1,60 @@
+name: ABI checks
+
+on:
+  push:
+    branches: [ master ]
+  pull_request:
+    branches: [ master ]
+
+env:
+  SRT_BASE: v1.5.0
```
Comment (maxsharabayko) [inline .github/workflows/abi.yml:10]: not used? Or needed to change value on the step "build"?

Comment (ethouris) [reply .github/workflows/abi.yml:10]: I wonder if there is any possible external file to provide the specific values. OTOH it will be rarely changed.

Comment (yomnes0) [reply .github/workflows/abi.yml:10]: It's the first way that came to my mind to transfer the tag version to the checkout action. It seems like "traditional" env variables are not shared between steps, and I couldn't get the ref from checkout@v3 to work properly by directly parsing the version there

Comment (ethouris) [reply .github/workflows/abi.yml:10]: If not, then we better have a script to apply appropriate changes, or generate this file.

Comment (yomnes0) [reply .github/workflows/abi.yml:10]: Could be done by using a github secret that would be updated at each tag generation

**Pull request merged**